src/magics/hashing.h

```diff
@@ -0,0 +1,95 @@
+
+#pragma once
+
+#include <array>
+#include <cstdint>
+
+namespace magic::hashing {
```

Comment:
Nitpick: We will probably add bishops here as well? Then we might want to namespace instead of prefix
```suggestion
namespace magic::hashing::rooks {
```